### PR TITLE
Policy Qualifiers other than id-qt-cps are no longer allowed as per CABF BRs

### PR DESCRIPTION
**Pull request by XolphinMartijn**
As of SC-62, only the cps policy qualifier is allowed in any certificate type.

### CODE REVIEW

**Comment by XolphinMartijn** [comment]
While I originally set out to search for User Notice, in fact all qualifiers are now a MUST NOT, except for id-qt-cps. PR and lint description updated to reflect proper language

**Review by zakird — CHANGES_REQUESTED** [review]
The description and lint code looks good. Can you update the name of the lint `e_user_notice_not_permitted` (in both code and file name) to match the new description of what's being checked?

**Comment by XolphinMartijn** [comment]
> The description and lint code looks good. Can you update the name of the lint `e_user_notice_not_permitted` (in both code and file name) to match the new description of what's being checked?

Done. Thank you for that suggestion. Likewise, function names have been updated

**Review by zakird — APPROVED** [review]

**Comment by zakird** [comment]
Looks good to me. @christopher-henderson do you want to quickly  👀  look over, or good to merge from your POV?

File: v3/testdata/policyQualifiersOtherThanCpsNotPermittedError.pem [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            X509v3 Extended Key Usage: 
+                TLS Web Client Authentication, TLS Web Server Authentication
+            X509v3 Basic Constraints: critical
+                CA:TRUE
+            X509v3 Authority Key Identifier: 
+                01:02:03
+            X509v3 Certificate Policies: 
+                0..F0.. ..g.....0...0...+.......0.0.....+.......0..0
..0.............An explicitText field includes the textual statement directly in the certificate.  The explicitText field is a string with a maximum size of 200 characters.  Conforming CAs SHOULD use the UTF8String encoding for explicitText.  0.....*...0...0...+.......0.0.....+.......0..0
..0.............An explicitText field includes the textual statement directly in the certificate.  The explicitText field is a string with a maximum size of 200 characters.  Conforming CAs SHOULD use the UTF8String encoding for explicitText.
```
Comment (christopher-henderson) [inline v3/testdata/policyQualifiersOtherThanCpsNotPermittedError.pem:32]: Clever use of the test certificate to also document what the certificate is testing :wink:

**Review by christopher-henderson — APPROVED** [review]
Looks clean, thank you very much @XolphinMartijn!

File: v3/testdata/policyQualifiersOtherThanCpsNotPermittedError.pem (hunk 1 above)
Comment (XolphinMartijn) [reply v3/testdata/policyQualifiersOtherThanCpsNotPermittedError.pem:32]: Noted for my next time!

**Pull request merged**